packages/k2-alpine/src/components/Avatar/HexagonImageView.tsx

```diff
@@ -0,0 +1,209 @@
+import MaskedView from '@react-native-masked-view/masked-view'
+import React, { useEffect, useState } from 'react'
+import { ImageSourcePropType, Platform, ViewStyle } from 'react-native'
+import Animated, {
+  Easing,
+  interpolateColor,
+  useAnimatedProps,
+  useAnimatedStyle,
+  useSharedValue,
+  withDelay,
+  withRepeat,
+  withTiming
+} from 'react-native-reanimated'
+import Svg, { Path } from 'react-native-svg'
+import { Image } from 'expo-image'
+import { alpha } from '../../utils'
+import {
+  colors,
+  darkModeColors,
+  lightModeColors
+} from '../../theme/tokens/colors'
+import { useTheme } from '../..'
+
+export const HexagonImageView = ({
+  source,
+  height,
+  backgroundColor,
+  isSelected,
+  hasLoading = false
+}: {
+  source: ImageSourcePropType
+  height: number
+  backgroundColor: string
+  isSelected?: boolean
+  hasLoading?: boolean
+}): JSX.Element => {
+  const { theme } = useTheme()
+  const selectedAnimation = useSharedValue(0)
+  const selectedAnimatedStyle = useAnimatedStyle(() => ({
+    opacity: selectedAnimation.value
+  }))
+  const [isLoading, setIsLoading] = useState(false)
+
+  useEffect(() => {
+    selectedAnimation.value = withTiming(isSelected ? 1 : 0, {
+      duration: 200,
+      easing: Easing.inOut(Easing.ease)
+    })
+  }, [isSelected, selectedAnimation])
+
+  return (
+    <MaskedView
+      maskElement={
+        <Svg width={height} height={height} viewBox={hexagonPath.viewBox}>
+          <Path d={hexagonPath.path} fill="black" />
+        </Svg>
+      }>
+      <Image
+        key={`image-${source}`}
+        resizeMode="cover"
+        source={source}
+        style={{ width: height, height: height, backgroundColor }}
+        onLoadStart={() => {
+          if (hasLoading) setIsLoading(true)
+        }}
```

Comment:
let's move this out and use a constant `noop` function. right now we are recreating onLoadStart on every render which will re-render Image